### PR TITLE
(Feature) New style for xDai network

### LINKED ISSUE #97: (Update) New style for xDai Network
We need to have separate colors and logo for xDai network in Validators DApp. Example (yellow-orange color scheme): https://blockscout.com/poa/dai

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
Closes #97 

**Description:**

Added support for xDai Network's styles.

Updated styles for the rest of the networks.

Added some standard components taken from other POA Network's dApps, in hopes of having a more coherent and consistent ecosystem.

**Some screenshots:**

screen shot 2019-01-16 at 17 22 14
screen shot 2019-01-16 at 17 22 24
screen shot 2019-01-16 at 17 26 33
screen shot 2019-01-16 at 17 26 44
screen shot 2019-01-16 at 17 27 09
screen shot 2019-01-16 at 17 33 02
screen shot 2019-01-16 at 17 33 11

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -258,102 +259,101 @@ class App extends Component {
       onChange: this.onChangeAutoComplete,
       id: 'address'
     }
-    let loader = this.state.loading ? <Loading /> : ''
 
-    const isDaiNetwork = netId === helpers.netIdByName('dai')
+    let loader = this.state.loading ? <Loading /> : ''
```
Comment (vbaranov) [inline src/App.js]: Does `loader` object change after assignment? if no, let's change to `const`

File: src/components/Validator/index.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      isCompany,
+      lastName,
+      licenseId,
+      netId,
+      networkBranch,
+      physicalAddresses,
+      updatedDate
+    } = this.props
+
+    if (netId === helpers.netIdByName('dai') && !createdDate) {
```
Comment (vbaranov) [inline src/components/Validator/index.js]: `dai` and other networks names are meet in different places in the code. Let's define constants for them.

**Review by vbaranov — CHANGES_REQUESTED** [review]
- There is no visible loader when switching to *Pending changes* or *All* back
- Impossible to search by full name (to be truth, it is not the issue of this PR, but it would be nice if it could be fixed) screenshot 2019-01-17 at 14 28 00
- change warning text **The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask** to **The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask/Nifty Wallet**

**Comment by vbaranov** [comment]
@gabitoesmiapodo 

Also,

- if to open directly `/set` page, the loader is shown permanently screenshot 2019-01-17 at 15 07 34
- wrong loader color for `Sokol` network by clicking on *Update metadata* button screenshot 2019-01-17 at 15 09 00

**Comment by gabitoesmiapodo** [comment]
@vbaranov 

1) There is no visible loader when switching to Pending changes or All back -> Fixed.

2) Impossible to search by full name (to be truth, it is not the issue of this PR, but it would be nice if it could be fixed) -> I think the way this is implemented it can get quite complicated easily (I mean, it could be improved), but I believe it's working as expected now. Please check it out.

3) change warning text The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask to The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask/Nifty Wallet -> Changed it.

4) Does loader object change after assignment? if no, let's change to const -> Fixed.

5) dai and other networks names are meet in different places in the code. Let's define constants for them. -> Done.

6) if to open directly /set page, the loader is shown permanently -> Fixed.

7) wrong loader color for Sokol network by clicking on Update metadata button -> Fixed.

**Comment by varasev** [comment]
I will also review that.

**Comment by vbaranov** [comment]
@gabitoesmiapodo 

screenshot 2019-01-18 at 19 57 59

**Comment by vbaranov** [comment]
@gabitoesmiapodo 

Will you plan to address the problem above?

**Comment by gabitoesmiapodo** [comment]
@vbaranov Sorry, I was on vacation until yesterday.

Yes, I'm working on it.

Any other feedback from you or @varasev ?

**Comment by varasev** [comment]
@gabitoesmiapodo I didn't review this yet, but I'll do this after the error above is fixed.

**Comment by vbaranov** [comment]
>Any other feedback from you

Oh, thanks. No, as far as I can see this is the last issue blocking the merging from my perspective.

**Comment by gabitoesmiapodo** [comment]
@vbaranov 

It should be fixed now.

I think it had something to do with me not rendering unnecessary components when they were not needed (instead of just hiding them), so I changed that and now they are hidden / shown depending on the situation (as they were before I changed that).

**Comment by vbaranov** [comment]
@gabitoesmiapodo I am still experiencing this issue in https://github.com/poanetwork/poa-dapps-validators/pull/103/commits/a8db3ac471d13900a471998ca53c323e15eaf952

**Comment by gabitoesmiapodo** [comment]
@vbaranov 

OK, any leads about what could be causing it? I'm not an actual validator (and don't have a validator key) and can't test the problem in real conditions.

I'm not sure about what could be causing it, all required nine fields are there.

**Comment by vbaranov** [comment]
@gabitoesmiapodo 

**TL;DR**: There is no problem with this PR. This is the issue of setup for testing. Could you also revert the last commit? Not rendering is more preferable, than hiding.

I figured out what is going on. Actually, I am not a validator too, and I am using https://github.com/poanetwork/poa-test-setup to test the set of governance dapps. By default, it retrieves the latest version of [governance smart-contracts](https://github.com/poanetwork/poa-network-consensus-contracts) and is trying to bootstrap Sokol network (networkID = *77*). But, actually, real Sokol network uses the previous version of smart-contract, where `CreateMetadata` method accepts only 7 parameters. And this is why @varasev implemented a condition here https://github.com/poanetwork/poa-dapps-validators/commit/4e0bed64461c4c7d8c47bd75c2639d0ec8ac1ec4#diff-7aae9601c64b3824098bf15eead67f6eR66, that adds 2 additional parameters to `CreateMetadata` method only for xDai network. So, this is an issue with `poa-test-setup`, not with this update of validators dapp.

**Comment by varasev** [comment]
`ValidatorMetadata.createMetadata` method was changed for `xDai` network and now [it expects 9](https://github.com/poanetwork/poa-chain-spec/blob/dai/abis/ValidatorMetadata.abi.json#L398) parameters instead of 7. However, we have previous `ValidatorMetadata.createMetadata` deployed in Sokol and Core, so for these networks it expects [7 parameters](https://github.com/poanetwork/poa-chain-spec/blob/sokol/abis/ValidatorMetadata.abi.json#L60).

**Comment by varasev** [comment]
The corresponding changes were introduced into `Validators DApp` in https://github.com/poanetwork/poa-dapps-validators/pull/98 and https://github.com/poanetwork/poa-dapps-validators/pull/102.

**Comment by gabitoesmiapodo** [comment]
@vbaranov 

Okay, reverted the changes for this section.

I hope it's fine now.

**Review by vbaranov — APPROVED** [review]
LGTM

**Comment by varasev** [comment]
Please don't merge this yet, I'll also check.

**Comment by gabitoesmiapodo** [comment]
@varasev Sure, no problem.

**Review by varasev — APPROVED** [review]
